pod.go

```diff
@@ -347,8 +347,16 @@ func (podConfig *PodConfig) valid() bool {
 	return true
 }
 
+const (
+	// R/W lock
+	exclusiveLock = syscall.LOCK_EX
+
+	// Read only lock
+	sharedLock = syscall.LOCK_SH
+)
+
 // lock locks any pod to prevent it from being accessed by other processes.
-func lockPod(podID string) (*os.File, error) {
+func lockPod(podID string, lockType int) (*os.File, error) {
```

Comment:
I'd prefer to see wrappers defined around a `__lockPod(podID string lockType)`. Something like:

```Go
func rLockPod(podID string) (*os.File, error) {
     return __lockPod(podID, sharedLock)
}

func rwLockPod(podID string) (*os.File, error) {
     return __lockPod(podID, exclusiveLock)
}
```

Reply:
Done